pywbemtools/pywbemcli/_display_cimobjects.py

```diff
@@ -478,10 +478,45 @@ def _display_instances_as_table(insts, table_width, table_format,
     else:
         max_cell_width = table_width
 
-    header_line = []
+    # TODO: Decide whether and how the showing of units should be controlled.
+    show_units = True
+
+    if show_units:
+        # Retrieve the most derived class of these instances. This is necessary
+        # because that is the only class that has all properties that may be
+        # needed.
```

Comment:
Second thoughts on the implementation: The algorithm to determine a most derived class is not covering cases where multiple leaf classes are involved. Each of them could add new properties with PUnit/Units qualifiers, so that no single class exists that has all of them.

Reply:
Agreed

Reply:
Why not build the  table of instances and supplement a all_properties list with the new properties introduced by each instance if that instance has properties not already in the table.  It is more expensive but we are dealing with thousands of instances for this display.

Reply:
We need the properties of the classes, not of the instances. But yes, something like that.

Reply:
DONE - Changed the implementation to retrieve a cache of all creation classes of the instances, which should normally be a short list, and in the loop through the properties, they are inspected for the qualifiers.